### PR TITLE
Remove constraint on dependencies version

### PR DESCRIPTION
**Pull request by xadupre**
The setup requires to install too old versions of dependencies. This PR removes these constraints.

### CODE REVIEW

**Comment by ksaur** [comment]
Hi @xadupre !  We pinned these to specific versions due to incompatibilities, and are trying to fix them as needed.  Are you blocked on a particular version of something?

**Comment by ksaur** [comment]
See also #454

**Comment by xadupre** [comment]
> Are you blocked on a particular version of something?

I'm refreshing onnxmltools CI which involves hummingbird-ml. More recent versions would be better. Right now I need to do `pip install hummingbird-ml --no-deps` to avoid CI to revert to older versions of scikit-learn, lightgbm. The errors I saw in this PR are very similar to what I had when updated sklearn-onnx for scikit-learn 0.24. I'll try to see if similar changes can fix the pipeline unless somebody is already working on it.

**Comment by ksaur** [comment]
> I'll try to see if similar changes can fix the pipeline unless somebody is already working on it.

Thanks for you looking into this, no one is currently looking at it!  We would greatly appreciate your help!

The information I have about our known version issues is as follows:
* For  scikit-learn, there was some issue with gradient boosting, i think you might have just re-discovered :)
* For onnxruntime (#454), we haven't dug into much yet.
* For lightgbm, there was some issue (#389) I think related to onnxmltools.
* For XGB, there is an outstanding issue (#495) due to a breaking change in April. 

Please let me know if you have questions and we can try to sort things out on our end as well!

**Comment by xadupre** [comment]
I may be able to fix all issues but this PR won't fix #389. I did a fix for this one in onnxmltools (https://github.com/onnx/onnxmltools/blob/master/onnxmltools/convert/lightgbm/operator_converters/LightGbm.py#L423). I'll see if I can easily do something similar.

**Comment by xadupre** [comment]
scikit-learn and XGB are done.

File: tests/test_lightgbm_converter.py [hunk 1]
```diff
@@ -225,8 +225,9 @@ def test_float64_lgbm_perf_tree_trav_regressor_converter(self):
         self._run_float64_lgbm_regressor_converter(1000, extra_config={"tree_implementation": "perf_tree_trav"})
 
     # Random forest in lgbm
+    @unittest.skipIf(True, reason="boosting_type=='rf' produces different probabilites.")
```
Comment (ksaur) [inline tests/test_lightgbm_converter.py:235]: Will this test ever be used?  Is this due to the runtime error added on lightgbm.py:69-72? If so, can you please add a comment?

File: hummingbird/ml/_parse.py [hunk 2]
```diff
@@ -624,6 +624,8 @@ def _parse_onnx_single_operator(topology, operator):
 
     # Add the operator in the topology.
     alias = get_onnxml_api_operator_name(operator.op_type)
+    if alias is None:
+        alias = "ONNXML" + operator.op_type
```
Comment (ksaur) [inline hummingbird/ml/_parse.py:628]: I am curious, in what cases does this happen?

**Review by ksaur — APPROVED** [review]
This is wonderful, thank you so much @xadupre!

File: hummingbird/ml/operator_converters/onnx/tree_ensemble.py [hunk 3]
```diff
@@ -40,6 +40,9 @@ def _get_tree_infos_from_onnx_ml_operator(model):
             threshold = attr.floats
         elif attr.name == "class_weights" or attr.name == "target_weights":
             values = attr.floats
+            if len(values) == 0:
+                raise TypeError(
+                    "Type mismatch with attribute %r." % attr)
```
Comment (interesaaat) [inline hummingbird/ml/operator_converters/onnx/tree_ensemble.py]: Nit: in HB we use `string.format()` instead of `%`. Cold you please use format for consistency?

**Review by interesaaat — APPROVED** [review]

File: hummingbird/ml/_parse.py (hunk 2 above)
Comment (xadupre) [reply hummingbird/ml/_parse.py:628]: StandardScaler is converted into an onnx Scaler when types are float. This operator only supports float, not double. So for double, the scaler is replaced by operator Sub, Mul. Alias is None is this case. The fix I did may be not the best one but it worked in this case.

File: hummingbird/ml/operator_converters/onnx/tree_ensemble.py (hunk 3 above)
Comment (xadupre) [reply hummingbird/ml/operator_converters/onnx/tree_ensemble.py]: done

File: tests/test_lightgbm_converter.py (hunk 1 above)
Comment (xadupre) [reply tests/test_lightgbm_converter.py:235]: I let the test because I assume this case will be supported. I added a comment in that way. I can remove the test.

Comment (ksaur) [reply tests/test_lightgbm_converter.py:235]: a comment is perfect, thanks!

**Comment by interesaaat** [comment]
Thanks @sdpython for the contribution!
